### PR TITLE
15740_fix: magic link dialog flow

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/15740

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 1]
```diff
@@ -672,52 +680,65 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
       const decodedToken = Base64.decode(base64TokenObject)
       const token = JSON.parse(decodedToken)
       const legalName = Base64.decode(this.base64OrgName)
+      // switch to match the account in the magic link
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:683]: This should be in a function itself

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 2]
```diff
@@ -672,52 +680,65 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
...
+      if (currentOrgId !== Number(this.orgId)) {
+        this.setCurrentAccountSettings({
+          id: Number(this.orgId),
+          label: legalName,
+          type: 'ACCOUNT',
+          urlpath: '',
+          urlorigin: ''
+        })
+        try {
+          await this.syncCurrentAccount(this.currentAccountSettings)
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:694]: Need some notes to rework these into Pinia for Vue3

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 3]
```diff
@@ -672,52 +680,65 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
...
       const response = await AffiliationInvitationService.acceptInvitation(invitationId, base64Token)
 
+      // 3. Adding magic link success
+      if (response.status === 200) {
+        this.showAddSuccessModalByEmail(identifier)
+      }
+    } catch (error) {
       // 4. Unauthorized
-      if (response.status === 401) {
+      if (error.response && error.response.status === 401) {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:732]: `error.response?.status == 401`

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 4]
```diff
@@ -672,52 +680,65 @@ export default class EntityManagement extends Mixins(AccountMixin, AccountChange
...
         this.showAuthorizationErrorModal()
         return
       }
-
-      // 5. Adding magic link success
-      if (response.status === 200) {
-        this.showAddSuccessModalByEmail(identifier)
+      // 5. Expired
+      if (error.response && error.response.status === 400 && error.response.data.code === MagicLinkInvitationStatus.EXPIRED_AFFILIATION_INVITATION) {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:737]: `error.response?.status === 400` and `error.response?.data.code`

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 5]
```diff
@@ -588,13 +593,16 @@ import { appendAccountId } from 'sbc-common-components/src/util/common-util'
...
+    ...mapActions(useOrgStore, ['syncAddress']),
+    ...mapActionsVuex('account', ['syncCurrentAccount'])
   }
 })
 export default class EntityManagement extends Mixins(AccountMixin, AccountChangeMixin, NextPageMixin) {
   @Prop({ default: '' }) readonly orgId: string
   @Prop({ default: '' }) readonly base64Token: string
   @Prop({ default: '' }) readonly base64OrgName: string
+  @Action(useOrgStore) protected addOrgSettings!: (org: Organization) => Promise<UserSettings>
+  readonly syncCurrentAccount!: (userSettings: AccountSettings) => Promise<AccountSettings>
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:605]: Should be marked remove Vuex for Pinia with Vue3 upgrade

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2513-8ylo5wr0.web.app

**Review by seeker25 — APPROVED** [review]

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2513-8ylo5wr0.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2513-8ylo5wr0.web.app

**Comment by Jxio** [comment]
waiting for more reviews, will merge at end of today, the linting should be passed, the error of unused reference actually used at code.

**Pull request merged**